proj/src/main/webapp/signup.js

```diff
@@ -143,15 +164,35 @@ function initializePasswordValidation(){
             passconf.classList.add("non-matching-pass");
             noMatch.hidden = false;
         }
-        if (meetRequirements(passVal) && noMatch.hidden) {
-            btnSignUp.disabled = false;
-        } else {
-            // disable sign up button
-            btnSignUp.disabled = true;
-        }
+        enableSignUp(passVal);
     })
 }
 
+function hasTag(){
+    const tagBox = document.getElementById("tags");
+    if (tagBox.value.trim().length != 0){
+        $('#tags').css('border','none');
+        return true;
+    } else {
+        // the tag box will only be in red after the user has input a valid password 
+        // this is to prevent the page from being visually overwhelming and help guide the user
+        if (noMatch.hidden){
+            $('#tags').css('border','3px solid #fa8072');
+        }
+        return false;
+    }
+}
+
+function enableSignUp(passVal){
+    $('#tags').keyup(hasTag);
```

Comment:
Since hasTag is being called on the next line, and enableSignUp is already called in the keyUp function for #tags,#pass,#passconf, we don't need this line. If this is here, then everytime we press a key, we'll add a new listener and hastags will get called an extra time unnecessarily.

Reply:
bump this comment.

Reply:
sorry, i missed this comment before, i have changed it